bigtable/filter_samples.rb

```diff
@@ -0,0 +1,217 @@
+  filter = Google::Cloud::Bigtable::RowFilter.value "PQ2A.*$"
+  read_with_filter project_id, instance_id, table_id, filter
+end
+
+# [END bigtable_filters_limit_value_regex]
+# [START bigtable_filters_limit_timestamp_range]
+def filter_limit_timestamp_range project_id, instance_id, table_id
+  timestamp_minus_hr = (Time.now.to_f * 1_000_000).round(-3) - 60 * 60 * 1000 * 1000
+  puts timestamp_minus_hr
+  filter = Google::Cloud::Bigtable::RowFilter.timestamp_range from: 0, to: timestamp_minus_hr
+
+  read_with_filter project_id, instance_id, table_id, filter
+end
+
+# [END bigtable_filters_limit_timestamp_range]
+# [START bigtable_filters_limit_block_all]
+def filter_limit_block_all project_id, instance_id, table_id
+  filter = Google::Cloud::Bigtable::RowFilter.block
+  read_with_filter project_id, instance_id, table_id, filter
+end
+
+# [END bigtable_filters_limit_block_all]
+# [START bigtable_filters_limit_pass_all]
+def filter_limit_pass_all project_id, instance_id, table_id
+  filter = Google::Cloud::Bigtable::RowFilter.pass
+  read_with_filter project_id, instance_id, table_id, filter
+end
+
+# [END bigtable_filters_limit_pass_all]
+# [START bigtable_filters_modify_strip_value]
+def filter_modify_strip_value project_id, instance_id, table_id
+  filter = Google::Cloud::Bigtable::RowFilter.strip_value
+  read_with_filter project_id, instance_id, table_id, filter
+end
+
+# [END bigtable_filters_modify_strip_value]
+# [START bigtable_filters_modify_apply_label]
+def filter_modify_apply_label project_id, instance_id, table_id
+  filter = Google::Cloud::Bigtable::RowFilter.label "labelled"
+  read_with_filter project_id, instance_id, table_id, filter
+end
+
+# [END bigtable_filters_modify_apply_label]
+# [START bigtable_filters_composing_chain]
+def filter_composing_chain project_id, instance_id, table_id
+  filter = Google::Cloud::Bigtable::RowFilter.chain.cells_per_column(1).family("cell_plan")
+  read_with_filter project_id, instance_id, table_id, filter
+end
+
+# [END bigtable_filters_composing_chain]
+def filter_composing_interleave project_id, instance_id, table_id
+  filter = Google::Cloud::Bigtable::RowFilter.interleave.value("true").qualifier("os_build")
+  read_with_filter project_id, instance_id, table_id, filter
+end
+
+# [END bigtable_filters_composing_interleave]
+# [START bigtable_filters_composing_condition]
+
+def filter_composing_condition project_id, instance_id, table_id
+  filter = Google::Cloud::Bigtable::RowFilter.condition(
+    Google::Cloud::Bigtable::RowFilter.chain.qualifier("data_plan_10gb").value("true")
+  )
+                                             .on_match(Google::Cloud::Bigtable::RowFilter.label("passed-filter"))
+                                             .otherwise(Google::Cloud::Bigtable::RowFilter.label("filtered-out"))
+  read_with_filter project_id, instance_id, table_id, filter
+end
+
+# [END bigtable_filters_composing_condition]
+
+
+# [START bigtable_filters_limit_row_sample]
```

Comment:
Why are these snippets included in so many region tags? Do they do everything the region tag implies they do?

Reply:
So I'm doing this to avoid duplicating the code to read data with a filter and print each row. By using multiple region tags, it will let me include that part in each part of the documentation. The double region tags will get excluded with a regex in the includecode for the documentation.